src/app/EventManagement.cpp

```diff
@@ -458,7 +458,7 @@ CHIP_ERROR EventManagement::CopyAndAdjustDeltaTime(const TLVReader & aReader, si
     // right after the priority to keep tags ordered
     if (aReader.GetTag() == TLV::ContextTag(EventDataElement::kCsTag_PriorityLevel))
     {
-        if (ctx->mpContext->mFirst)
+        if (ctx->mpContext->mFirst || (ctx->mpContext->mCurrentEventNumber - ctx->mpContext->mPreviousEventNumber > 1))
```

Comment:
Do we have tests checking that:

1. If an event number is skipped the next event has a number.
2. If an event number is not skipped, we elide things.
3. If an even number is initially not skipped but then eviction evicts from the middle (due to being lower priority) we correctly have an event number after the skip.

Though does the spec even guarantee us that events will appear in event number order?  It seems like they are grouped by event path and only by event number within the path, per "Incoming Read Request and Subscribe Request Action Processing".  @mrjerryjohns is that expected?

Reply:
https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Encoding-Specification.adoc#examples-6
See this example, they are not actually "grouped" by event path, if third event has the different path as the second event, 
--third event can omit event number if their number difference is 1,  otherwise, it would show event number(implemented in this PR)
--except first, second/third... event can use delta time（implemented) for compression purpose

if second event has the same path as the first event, 
--Can omit priority level and path(not implemented)

Reply:
> See this example, they are not actually "grouped" by event path

I am looking at https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Interaction-Model.adoc#incoming-read-request-and-subscribe-request-action-processing the bullet point about "All concrete paths in EventRequests".

That is, what is the ordering of the event records that are passed to the encoding layer to start with?

Reply:
You're right @bzbarsky-apple in that the spec currently has a primary sort on the event path and type, and a secondary sort on event number.

We should remedy that to be a primary sort on event number, filtered by event path/type.

Reply:
thanks, filed the spec ticket, https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3402